cli/command_repository_sync.go

```diff
@@ -303,19 +302,25 @@ func (c *commandRepositorySyncTo) syncCopyBlob(ctx context.Context, m blob.Metad
 		return errors.Wrapf(err, "error reading blob '%v' from source", m.BlobID)
 	}
 
-	if err := dst.PutBlob(ctx, m.BlobID, data.Bytes(), blob.PutOptions{}); err != nil {
-		return errors.Wrapf(err, "error writing blob '%v' to destination", m.BlobID)
+	opt := blob.PutOptions{}
+	if c.repositorySyncTimes {
+		opt.SetModTime = m.Timestamp
 	}
 
-	if c.repositorySyncTimes {
-		if err := dst.SetTime(ctx, m.BlobID, m.Timestamp); err != nil {
-			if errors.Is(err, blob.ErrSetTimeUnsupported) {
-				c.setTimeUnsupportedOnce.Do(func() {
-					log(ctx).Infof("destination repository does not support setting time")
-				})
-			}
+	if err := dst.PutBlob(ctx, m.BlobID, data.Bytes(), opt); err != nil {
+		if errors.Is(err, blob.ErrSetTimeUnsupported) {
+			// run again without SetModTime, emit a warning
+			opt.SetModTime = time.Time{}
+
+			log(ctx).Warnf("destination repository does not support preserving modification times")
+
+			c.repositorySyncTimes = false
```

Comment:
nit; nothing t todo here, just a mention

could be "undefined behavior" for a bit but should be good eventually ...